File: examples/python/subject_read_workflow.py
import json
import sys
from urllib.parse import quote

from memside_http import api_request


subject_id = sys.argv[1] if len(sys.argv) > 1 else None
if not subject_id:
    subjects = api_request("GET", "/subjects", query={"limit": 1})
    subject_id = subjects[0]["id"] if subjects else None

if not subject_id:
    print("No accessible Subjects found.")
    raise SystemExit(0)

encoded_id = quote(subject_id, safe="")
result = {
    "subject": api_request("GET", f"/subjects/{encoded_id}"),
    "memories": api_request(
        "GET",
        f"/subjects/{encoded_id}/memories",
        query={"limit": 10},
    ),
    "context": api_request("GET", f"/subjects/{encoded_id}/context"),
    "facts": api_request(
        "GET",
        f"/subjects/{encoded_id}/facts",
        query={"limit": 10},
    ),
    "insights": api_request(
        "GET",
        f"/subjects/{encoded_id}/memory-insights",
        query={"limit": 10},
    ),
}

print(json.dumps(result, indent=2))

File: examples/javascript/subject-read-workflow.mjs
import { memside } from "./memside-client.mjs";

let subjectId = process.argv[2];
if (!subjectId) {
  const subjects = await memside("/subjects?limit=1");
  subjectId = subjects[0]?.id;
}

if (!subjectId) {
  console.log("No accessible Subjects found.");
  process.exit(0);
}

const encodedId = encodeURIComponent(subjectId);
const [subject, memories, context, facts, insights] = await Promise.all([
  memside(`/subjects/${encodedId}`),
  memside(`/subjects/${encodedId}/memories?limit=10`),
  memside(`/subjects/${encodedId}/context`),
  memside(`/subjects/${encodedId}/facts?limit=10`),
  memside(`/subjects/${encodedId}/memory-insights?limit=10`),
]);

console.log(
  JSON.stringify({ subject, memories, context, facts, insights }, null, 2)
);
